app/assets/javascripts/discourse/controllers/password-reset.js.es6

```diff
@@ -45,7 +47,17 @@ export default Ember.Controller.extend(PasswordValidation, {
             DiscourseURL.redirectTo(result.redirect_to || '/');
           }
         } else {
-          if (result.errors && result.errors.password && result.errors.password.length > 0) {
+          if (result.errors && result.errors.second_factor) {
+            this.set('secondFactorRequired',true);
+            this.set('password', null);
+            this.set('errorMessage', 'Invalid token');
+          }
+          else if (this.get('secondFactorRequired') === true) {
```

Comment:
Minor but `this.get('secondFactorRequired')` should be sufficient here without `=== true`